src/sources/demo_logs.rs

```diff
@@ -242,25 +260,33 @@ impl_generate_config_from_default!(DemoLogsConfig);
 #[typetag::serde(name = "demo_logs")]
 impl SourceConfig for DemoLogsConfig {
     async fn build(&self, cx: SourceContext) -> crate::Result<super::Source> {
+        let log_namespace = cx.log_namespace(self.log_namespace);
+
         self.format.validate()?;
-        let decoder = DecodingConfig::new(
-            self.framing.clone(),
-            self.decoding.clone(),
-            LogNamespace::Legacy,
-        )
-        .build();
+        let decoder =
+            DecodingConfig::new(self.framing.clone(), self.decoding.clone(), log_namespace).build();
         Ok(Box::pin(demo_logs_source(
             self.interval,
             self.count,
             self.format.clone(),
             decoder,
             cx.shutdown,
             cx.out,
+            log_namespace,
         )))
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(self.decoding.output_type())]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        // There is a global and per-source `log_namespace` config. The source config overrides the global setting,
+        // and is merged here.
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+
+        let schema_definition = self
+            .decoding
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata();
+
+        vec![Output::default(self.decoding.output_type()).with_schema_definition(schema_definition)]
```

Comment:
Could be off here but- wouldn't this implementation of `outputs` be very similar in the other sources and if so, would it make sense to extract that to reduce duplicated code ?

Reply:
Many sources will contain several calls to `.with_source_metadata(..)` which will be unique to that source. This just happens to be a really simple source. (See the `datadog_agent` which is more complicated).

Other than that, there will be a few lines duplicated, but I'm not seeing a great way to deduplicate that without potentially making it harder to read / configure.